chia/wallet/puzzles/dao_treasury.clvm

```diff
@@ -208,14 +213,19 @@
         )
       )
       ; this is the add money spend case
+      ; COMMENT: This should probably be 0 or something configurable to prevent spaming the add money case a DOS
       (if (> new_amount_change -1)
         (merge_list
+          ; COMMENT: maybe this should be a loop so you can do more than one p2_singleton at once
           (if proposal_innerpuz_or_optional_p2_singleton_coin_id
+            ; COMMENT: An attacker here could give you a p2_singleton to this DAO, then specify an amount increase of 0 or 1 and melt the p2_singleton's value without actually giving it to the DAO
             (list
               (list CREATE_PUZZLE_ANNOUNCEMENT proposal_innerpuz_or_optional_p2_singleton_coin_id)
+              ; COMMENT: Using the existing p2_singleton here pretty much eliminates support for CATs or NFTs being sent to the DAO
```

Comment:
This isn't using a rigid p2_singleton mod, it's taking a coin ID in from the solution and just asserts that it is making the `$` announcement